src/main/kotlin/org/rust/lang/utils/RsDiagnostic.kt

```diff
@@ -1217,14 +1217,30 @@ sealed class RsDiagnostic(
             )
         }
     }
+
+    class NonConstantValueInConstantError(element: PsiElement) : RsDiagnostic(element) {
+        override fun prepare() = PreparedAnnotation(
+            ERROR,
+            E0435,
+            "A non-constant value was used in a constant expression"
+        )
+    }
+
+    class NonConstantFunctionCallInConstantError(element: PsiElement) : RsDiagnostic(element) {
```

Comment:
```suggestion
    class NonConstantCallInConstantError(element: PsiElement) : RsDiagnostic(element) {
```

This diagnostic can be applied not only for function calls but also at least for method calls as well. Let's make its name more generic